code/game/machinery/bots/draculabot.dm

```diff
@@ -0,0 +1,278 @@
+								/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje,
+								/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje,
+								/obj/item/weapon/reagent_containers/food/snacks/chococoin/wrapped,
+								/obj/item/weapon/reagent_containers/food/snacks/chococoin/wrapped,
+								/obj/item/weapon/reagent_containers/food/snacks/chococoin/wrapped,
+								/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje_helper_dummy) //3/7 chance for one cookie, 3/7 chance for chococoin, 1/7 for many cookies!
+	var/list/contained_bags = list()
+	var/obj/item/weapon/reagent_containers/blood/last_bag
+
+/obj/machinery/bot/bloodbot/New()
+	..()
+	for(var/i = 1 to 8)
+		contained_bags += new /obj/item/weapon/reagent_containers/blood/empty(src)
+
+/obj/machinery/bot/bloodbot/Destroy()
+	for(var/obj/O in contained_bags)
+		O.forceMove(get_turf(src))
+	contained_bags = null
+	last_bag = null
+	..()
+
+/obj/machinery/bot/bloodbot/update_icon()
+	if(contained_bags.len || emagged)
+		icon_state = "bloodbot[emagged][currently_drawing_blood]"
+	else
+		icon_state = "bloodbot-e"
+
+/obj/machinery/bot/bloodbot/turn_on()
+	..()
+	update_icon()
+	updateUsrDialog()
+
+/obj/machinery/bot/bloodbot/turn_off()
+	if(emagged)
+		return
+	..()
+	currently_drawing_blood = 0
+	update_icon()
+
+/obj/machinery/bot/bloodbot/attack_paw(mob/user as mob)
+	return attack_hand(user)
+
+/obj/machinery/bot/bloodbot/attack_hand(mob/user as mob)
+	if(..())
+		return
+	var/dat
+	dat += "<TT><B>Acula-class Blood Donation Bot v1.0</B></TT><BR><BR>"
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/datum/reagent/blood/B = H.get_blood(H.vessel)
+		if(B.data["virus2"])
+			dat += "<span class='danger'>WARNING: Viral agent detected. Ineligible for blood donation.</span><BR>"
+		else
+			dat += {"Welcome [H]! Your blood level is [round(B.volume/560*100)]%, and your blood type is [B.data["blood_type"]].<BR>
+				You must have at least [round(509/560*100)]% to donate blood. <a href='?src=\ref[src];donate=[1]'>Donate now!</a><BR>
+				The next reward will be dispensed after [55 - since_last_reward] units of blood are donated.<BR>"}
+	dat += {"Status: <A href='?src=\ref[src];power=1'>[src.on ? "On" : "Off"]</A><BR>
+			The maintenance panel is [src.open ? "opened" : "closed"]<BR>"}
+	if(!src.locked || issilicon(user))
+		dat += "<TT>Blood Storage:<BR>"
+		var/counter = 0
+		if(!contents.len)
+			dat += "There are no blood packs available.<BR>"
+		for (var/obj/item/weapon/reagent_containers/blood/E in contained_bags)
+			counter++
+			dat += "Slot [counter]: [E] ([100 * E.reagents.total_volume / E.reagents.maximum_volume]% filled)<A href='?src=\ref[src];slot=\ref[E]'>(Eject)</A><BR>"
+		dat += "</TT>The speaker switch is [src.quiet ? "off" : "on"]. <a href='?src=\ref[src];togglevoice=[1]'>Toggle</a><br>"
+
+	dat = jointext(dat,"")
+	var/datum/browser/popup = new(usr, "bloodbot", "[name]", 575, 400, src)
```

Comment:
Make the UI key (`"bloodbot"`) be `\ref[src]` instead. No issues having two UIs open at once!